packages/ckb-indexer/src/collector.ts

```diff
@@ -43,82 +43,117 @@ export class CKBCellCollector implements BaseCellCollector {
       bufferSize: undefined,
     };
     this.queries = { ...defaultQuery, ...this.queries };
-    if (
-      !this.queries.lock &&
-      (!this.queries.type || this.queries.type === "empty")
-    ) {
+    this.validateParams(this.queries);
+    this.convertParams();
```

Comment:
```suggestion
    this.validateQueryOption(this.queries);
    this.convertQueryOptionToSearchKey();
```

The method names here can be confusing, in the implementation of `@ckb-lumos/ckb-indexer` we should distinguish between `SearchKey`(for indexer) and `QueryOption`(for client)